projects/common/src/lib/user/user.service.ts

```diff
@@ -1,66 +1,215 @@
+                                queryParamsHandling: 'merge',
+                            });
+                        });
+                } else {
+                    // restore old session if possible
+                    this.sessionFromBrowserOrCreateGuest().subscribe({
+                        next: (session) => {
+                            this.session$.next(session);
+                        },
+                        error: (error) => {
+                            // only show error if we did not expect it
+                            if (error?.error?.error !== 'Unauthorized') {
+                                this.notificationService.error(error?.error?.message);
+                            }
+                            this.session$.next(undefined);
+                        },
+                    });
+                }
+            }
+            if (!status.available && this.sessionInitialized) {
+                this.sessionInitialized = false;
                 this.session$.next(undefined);
-            });
+                this.notificationService.error('Session close caused by backend shutdown');
+            }
+        });
+
+        // update backend info when backend is available
+        this.getBackendStatus().subscribe(async (status) => {
+            if (status.available) {
+                const info = await this.backendApi.serverInfoHandler();
+                this.backendInfo$.next(info);
+            }
+        });
 
         this.getSessionStream().subscribe({
-            next: (session) => this.userApi.next(new UserApi(apiConfigurationWithAccessKey(session.id))),
+            next: (session) => {
+                this.userApi.next(new UserApi(apiConfigurationWithAccessKey(session.sessionToken)));
+                this.sessionApi.next(new SessionApi(apiConfigurationWithAccessKey(session.sessionToken)));
+            },
         });
+
+        // update quota when session changes or update is triggered
+        this.createSessionQuotaStream();
+
+        this.triggerBackendStatusUpdate();
     }
 
-    getSessionTokenStream(): Observable<string> {
-        return this.getSessionStream().pipe(map((session) => session.id));
+    async triggerBackendStatusUpdate(): Promise<void> {
+        try {
+            const _available = await this.backendApi.availableHandler();
+
+            this.backendStatus$.next({available: true});
+        } catch (error) {
+            this.backendStatus$.next({available: false, httpError: error as HttpErrorResponse});
+        }
     }
 
-    getSessionTokenForRequest(): Observable<string> {
-        return this.getSessionTokenStream().pipe(first());
+    getBackendStatus(): Observable<BackendStatus> {
+        return this.backendStatus$;
+    }
+
+    sessionFromBrowserOrCreateGuest(): Observable<Session> {
+        return this.restoreSessionFromBrowser().pipe(
+            catchError((error) => {
+                console.error(error);
```

Comment:
we always get the nasty error here:
```
ResponseError: Response returned an error code
    ResponseError runtime.js:258
    request runtime.js:193
    fulfilled runtime.js:23
    Angular 31
    adopt runtime.js:17
    ZoneAwarePromise Angular
    adopt runtime.js:16
    step runtime.js:36
    __awaiter runtime.js:38
    ZoneAwarePromise Angular
    __awaiter runtime.js:20
    fetchApi runtime.js:101
    request runtime.js:189
    fulfilled runtime.js:23
    Angular 12
[user.service.ts:154:24](http://localhost:4200/projects/common/src/lib/user/user.service.ts)
``

Reply:
d436c32aa6b954a1e1b0715e343165c6908bb56d